### PR TITLE
PHP 8: clean up `Database` package

### PR DESCRIPTION
**Pull request by distantnative**

### CODE REVIEW

File: src/Database/Db.php [hunk 1]
```diff
@@ -171,12 +157,15 @@ public static function __callStatic(string $method, $arguments)
...
  */
-Db::$queries['column'] = function (string $table, string $column, $where = null, string $order = null, int $offset = 0, int $limit = null) {
+Db::$queries['column'] = function (
+	string $table,
+	string $column,
+	mixed $where = null,
+	string|null $order = null,
+	int $offset = 0,
+	int|null $limit = null
+) {
```
Comment (afbora) [inline src/Database/Db.php]:
```suggestion
): mixed {
```

File: src/Database/Db.php [hunk 2]
```diff
@@ -156,12 +141,13 @@ public static function __callStatic(string $method, $arguments)
...
- * @param int $limit
- * @return mixed
  */
-Db::$queries['first'] = Db::$queries['row'] = Db::$queries['one'] = function (string $table, $columns = '*', $where = null, string $order = null) {
+Db::$queries['first'] = Db::$queries['row'] = Db::$queries['one'] = function (
+	string $table,
+	mixed $columns = '*',
+	mixed $where = null,
+	string|null $order = null
+) {
```
Comment (afbora) [inline src/Database/Db.php]:
```suggestion
): mixed {
```

File: src/Database/Sql.php [hunk 3]
```diff
@@ -841,14 +764,8 @@ public function values(string $table, $values, string $separator = ', ', bool $s
...
 	 * Creates a list of fields and values
-	 *
-	 * @param string $table
-	 * @param string|array $values
-	 * @param string $separator
-	 * @param bool $enforceQualified
-	 * @param array
 	 */
-	public function valueList(string $table, $values, string $separator = ',', bool $enforceQualified = false): array
+	public function valueList(string $table, string|array $values, string $separator = ',', bool $enforceQualified = false): array
```
Comment (afbora) [inline src/Database/Sql.php]:
```suggestion
	public function valueList(
		string $table,
		string|array $values,
		string $separator = ',',
		bool $enforceQualified = false
	): array {
```

File: src/Database/Sql.php [hunk 4]
```diff
@@ -887,15 +804,8 @@ public function valueList(string $table, $values, string $separator = ',', bool
...
-	 *
-	 * @param string $table
-	 * @param string|array $values
-	 * @param string $separator
-	 * @param bool $enforceQualified
-	 * @param array
-	 * @return array
 	 */
-	public function valueSet(string $table, $values, string $separator = ',', bool $enforceQualified = false): array
+	public function valueSet(string $table, string|array $values, string $separator = ',', bool $enforceQualified = false): array
```
Comment (afbora) [inline src/Database/Sql.php]:
```suggestion
	public function valueSet(
		string $table, 
		string|array $values, 
		string $separator = ',', 
		bool $enforceQualified = false
	): array {
```

File: src/Database/Sql.php [hunk 5]
```diff
@@ -823,8 +741,13 @@ public function validateColumn(string $table, string $column): bool
...
 	 */
-	public function values(string $table, $values, string $separator = ', ', bool $set = true, bool $enforceQualified = false): array
-	{
+	public function values(
+		string $table,
+		mixed $values,
+		string $separator = ', ',
+		bool $set = true,
+		bool $enforceQualified = false
+	): array {
```
Comment (afbora) [inline src/Database/Sql.php:750]:
```suggestion
	public function values(
		string $table,
		mixed  $values,
		string $separator = ', ',
		bool   $set = true,
		bool   $enforceQualified = false
	): array
	{
```

This change (and followings) suggested with based on `.editorconfig` by PHPStorm IDE.

File: src/Database/Query.php [hunk 6]
```diff
@@ -280,9 +238,9 @@ public function primaryKeyName(string $primaryKeyName)
 	 * By default all columns will be selected
 	 *
 	 * @param mixed $select Pass either a string of columns or an array
-	 * @return \Kirby\Database\Query
+	 * @return $this
 	 */
-	public function select($select)
+	public function select(mixed $select): static
```
Comment (afbora) [inline src/Database/Query.php]: `$this->select` property type is `array|string|null` but this method accept as `mixed`? Not consistent.

File: src/Database/Query.php [hunk 7]
```diff
@@ -348,7 +306,7 @@ public function innerJoin($table, $on)
 	 * Sets the values which should be used for the update or insert clause
 	 *
 	 * @param mixed $values Can either be a string or an array of values
-	 * @return \Kirby\Database\Query
+	 * @return $this
 	 */
 	public function values($values = [])
```
Comment (afbora) [inline src/Database/Query.php]:
```suggestion
	public function values($values = []): static
```

File: src/Database/Query.php [hunk 8]
```diff
@@ -446,10 +402,9 @@ public function andWhere(...$args)
 	/**
 	 * Attaches a group by clause
 	 *
-	 * @param string|null $group
-	 * @return \Kirby\Database\Query
+	 * @return $this
 	 */
-	public function group(string $group = null)
+	public function group(string|null $group = null): static
```
Comment (afbora) [inline src/Database/Query.php:407]: `$this->group` type is `mixed` but this method accept as `string|null`. Not consistent.

File: src/Database/Query.php [hunk 9]
```diff
@@ -490,10 +444,9 @@ public function order(string $order = null)
 	/**
 	 * Sets the offset for select clauses
 	 *
-	 * @param int|null $offset
-	 * @return \Kirby\Database\Query
+	 * @return $this
 	 */
-	public function offset(int $offset = null)
+	public function offset(int|null $offset = null): static
```
Comment (afbora) [inline src/Database/Query.php]: `$this->offset` is not nullable.

File: src/Database/Query.php [hunk 10]
```diff
@@ -614,12 +552,9 @@ public function avg(string $column): float
 	 * Builds an aggregation query.
 	 * This is used by all the aggregation methods above
 	 *
-	 * @param string $method
-	 * @param string $column
 	 * @param int $default An optional default value, which should be returned if the query fails
-	 * @return mixed
 	 */
-	public function aggregate(string $method, string $column = '*', $default = 0)
+	public function aggregate(string $method, string $column = '*', $default = 0): mixed
```
Comment (afbora) [inline src/Database/Query.php]:
```suggestion
	public function aggregate(string $method, string $column = '*', int $default = 0): mixed
```

File: src/Database/Query.php [hunk 11]
```diff
@@ -901,21 +808,16 @@ public function update($values = null, $where = null)
 	 * Fires a delete query
 	 *
 	 * @param mixed $where You can pass a where clause here or set it with ->where() before
-	 * @return bool
 	 */
-	public function delete($where = null)
+	public function delete(mixed $where = null): bool
```
Comment (afbora) [inline src/Database/Query.php:812]: When `debug` enabled, `$this->execute()` method returns `array`. Is that important?

**Comment by distantnative** [comment]
messed up the branch, starting over

**Pull request closed without merging**